Cargo.toml

```diff
@@ -454,12 +454,13 @@ sources-splunk_hec = ["bytesize", "sources-utils-tls", "warp"]
 sources-statsd = ["listenfd", "sources-utils-tcp-keepalive", "sources-utils-tcp-socket", "sources-utils-tls", "sources-utils-udp", "sources-utils-unix", "tokio-util/net"]
 sources-stdin = ["bytesize"]
 sources-syslog = ["bytesize", "listenfd", "tokio-util/net", "sources-utils-udp", "sources-utils-tcp-keepalive", "sources-utils-tcp-socket", "sources-utils-tls", "sources-utils-unix", "syslog_loose"]
+sources-utils-decoding = []
```

Comment:
This mainly exists to not mark the common `src/sources/util/decoding` configuration code as dead.

I figured it would be nicer to let sources register here rather than explicitly listing those that use decoders in the source code as `#[cfg(any(feature = "..."))]`.

> decoder-related feature flags similarly to the existing source/transform/sink pattern than keep them nested under source utils.

Sounds good. One reason for not putting them on the top level would be that decoders only apply to sources. We could do it anyway to simplify the project structure.